### DISCUSSION TITLE
some  questions about Code writing

### DISCUSSION THREAD

**Discussion by cccc164** [gempy-project/gempy#965] [General]
I recently wanted to learn this method. When I tried to reproduce it in two dimensions, I found that it was mentioned in the paper 'GemPy 1.0: open-source stochastic geological modeling and inversion' that the partial derivative of r with respect to the hu direction is -hu/r, I would like to ask where this negative sign comes from?

**Comment by NilsChudalla** [comment]
Hi @cccc164 , 
I had a look at the paper and I must confess, that I could not make any sense of it either. I summon @Deepprakash222, because I have more confidence in his math skills than in mine :D the equation in question is in the appendix of the [paper](https://gmd.copernicus.org/articles/12/1/2019/gmd-12-1-2019.pdf), page 23, B6.

**Reply by cccc164** [reply]
At the same time, I don't quite understand how the original formula is given. I think if it is the common co-kriging form, that is, there is no X0 on the left side of the equation and no Xa' on the right side of the equation, we all agree with this formula. The potential field method modifies it into an incremental form, so why not subtract the same value from both sides of the equation, that is, subtract X0。

**Reply by cccc164** [reply]
Thank you for your reply！

**Reply by cccc164** [reply]
And I don't think this is a mistake in the paper, because I checked the code and found that it is also written like this.

**Reply by Deepprakash222** [reply]
That is a mistake in derivation written in the paper.  It should be positive as you have pointed it out. The reason, why this works in case of the code is because there is an extra negative terms appear in from of the expression which cancel out the negative of -hu/r and thus that becomes consistent.

**Reply by cccc164** [reply]
Could you please look at the formulas B18 and B19? I think that if there is a derivation error in these two formulas, the negative sign cannot be offset as in the above formulas.

**Reply by cccc164** [reply]
And for the Kriging formula given by this method, I don't quite understand why this equation can be established. I think it does not conform to the co-Kriging that we usually know.

**Reply by Deepprakash222** [reply]
[Kriging.pdf](https://github.com/user-attachments/files/18321745/Kriging.pdf)

**Reply by Deepprakash222** [reply]
I have derived all the formulation in the above pdf. you can have a look.

**Reply by cccc164** [reply]
Ok, I'll have a look, thanks a lot for your help!
